lib/services/consumed/status-bar/signal-list-view.js

```diff
@@ -50,7 +50,7 @@ export default class SignalListView {
       this.cancel();
     }
 
-    const kernel = store.kernel;
+    const kernel = this.kernel;
```

Comment:
This doesn't work if the kernel is not the one with which this `SignalListView` instance gets constructed.

Reply:
good catch, because of the handle kernel command. I will update that function to require a kernel arg

Reply:
Doesn't work:
- Launch two different kernels for two editors
- Click status bar tile inside one editor: Would show kernel 1
- Then switch the editor and click again the tile: Still shows kernel 2.

What you really should do is to update `kernel` of the `SignalListView` instance. It's not the talks about `handleKernelCommand` and thus I think you should revert the change on it.

Reply:
The reason it wasn't working was because I had a logic error when I ported the old code. It works now for real. Is their some other reason why you are against this change because it really is not ideal to use global variables everywhere

Reply:
Also when removing the logic error one option was to update the kernel, so I did that as suggested.

Reply:
I agree with the idea of avoiding to use the global variables, of course ! 😃 
My point was just to point out the logic inside `SignalListView` seemed to be broken, and it seemed to me like you unnecessarily changed `handleKernelCommand`.

I confirmed `handleKernelCommand` is just used within `SignalListView`, thus it would just be okay as far as the logic inside `SignalListView` works though.

Personally I want to complete things with minimal code change (just my preference), thus I recommended that you revert the change. So I shouldn't have said "you _should revert_" since it' may sound kinda too strong, sorry 😅